### PR TITLE
Adobe analytics implementation for iOS

### PR DESCRIPTION
**Pull request by jeremie-movify**
- Added native iOS implementation
- Added documentation
- Simplified plugin usage

### CODE REVIEW

File: ios/Classes/PluginMethod.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//
+//  PluginMethod.swift
+//
+//  Created by Kymer Gryson on 02/10/2019.
+//
+
+import Foundation
+
+enum PluginMethod {
```
Comment (Kymer) [inline ios/Classes/PluginMethod.swift]: YASS 👍  but these still seem to be unused in ios/Classes/SwiftFlutterAdobeExperiencePlatformPlugin.swift ?

**Review by Kymer — CHANGES_REQUESTED** [review]

File: example/lib/main.dart [hunk 2]
```diff
@@ -12,32 +10,15 @@ class MyApp extends StatefulWidget {
...
-    // If the widget was removed from the tree while the asynchronous platform
-    // message was in flight, we want to discard the reply rather than calling
-    // setState to update our non-existent appearance.
-    if (!mounted) return;
-
-    setState(() {
-      _platformVersion = platformVersion;
-    });
+  Future<void> trackState(String state, Map<String, String> data, {@required BuildContext context}) async {
+    final success = await AdobeExperiencePlatform.trackState(state, data);
```
Comment (timrijckaert) [inline example/lib/main.dart]: `AdobeExperiencePlatform` is too broad right?
We wanted to split this up in multiple extensions.
Shouldn't this be part of the `AdobeAnalytics` extension?

File: example/pubspec.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ publish_to: 'none'
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: ">=2.2.2 <3.0.0"
```
Comment (timrijckaert) [inline example/pubspec.yaml]: `2.5.0`

File: example/lib/main.dart (hunk 2 above)
Comment (timrijckaert) [reply example/lib/main.dart]: Never mind just saw the next PR

File: example/pubspec.yaml (hunk 3 above)
Comment (jeremie-movify) [reply example/pubspec.yaml]: Done 👍

**Comment by timrijckaert** [comment]
Can we drop the `Experience Platform` reference everywhere?
Considering the community they will probably look for `flutter adobe analytics`

File: packages/adobe_analytics/.idea/libraries/Dart_SDK.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<component name="libraryTable">
```
Comment (timrijckaert) [inline packages/adobe_analytics/.idea/libraries/Dart_SDK.xml]: don't check in .idea folder

**Review by timrijckaert — CHANGES_REQUESTED** [review]

File: ios/Classes/PluginMethod.swift (hunk 1 above)
Comment (jeremie-movify) [reply ios/Classes/PluginMethod.swift]: Indeed, that's now fixed ; thanks for pointing that out :-)

**Review by timrijckaert — APPROVED** [review]

File: packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(name: "adobe_analytics", binaryMessenger: registrar.messenger())
+    let instance = SwiftAdobeAnalyticsPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+  
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    do {
+      switch try PluginMethod(from: call) {
```
Comment (Kymer) [inline packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift:41]: <3

**Review by Kymer — APPROVED** [review]

**Pull request merged**